python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,519 @@
+    single-band and multi-band data. It defines the detection, deblending,
+    and measurement subtasks, which are intended to be executed in sequence
+    by the driver tasks.
+    """
+
+    idGenerator = measBase.DetectorVisitIdGeneratorConfig.make_field(
+        doc="Configuration for generating catalog IDs from data IDs consistently across subtasks."
+    )
+
+    detection = pexConfig.ConfigurableField(
+        target=measAlgorithms.SourceDetectionTask, doc="Subtask to detect sources in the image."
+    )
+
+    deblender = pexConfig.ChoiceField[str](
+        doc="Which deblender to use?",
+        default="meas_deblender",
+        allowed={
+            "meas_deblender": "Deblend using meas_deblender (only single-band)",
+            "scarlet": "Deblend using scarlet (single- or multi-band)",
+        },
+    )
+
+    deblend = pexConfig.ConfigurableField(
+        target=measDeblender.SourceDeblendTask, doc="Subtask to split blended sources into components."
+    )
+
+    measurement = pexConfig.ConfigurableField(
+        target=measBase.SingleFrameMeasurementTask,
+        doc="Subtask to measure sources and populate the output catalog",
+    )
+
+    def __setattr__(self, key, value):
+        """Intercept changes to 'deblender' and retarget subtask if needed."""
+        super().__setattr__(key, value)
+
+        # This is to ensure the deblend target is set correctly whenever the
+        # deblender is changed. This is required because `setDefaults` is not
+        # automatically invoked during reconfiguration.
+        if key == "deblender":
+            self._retargetDeblend()
+
+    def setDefaults(self):
+        super().setDefaults()
+        self._retargetDeblend()
+
+    def _retargetDeblend(self):
+        if self.deblender == "scarlet":
+            self.deblend.retarget(scarlet.ScarletDeblendTask)
+        elif self.deblender == "meas_deblender":
+            self.deblend.retarget(measDeblender.SourceDeblendTask)
+
+    def validate(self):
+        super().validate()
+        targetMap = {
+            "scarlet": scarlet.ScarletDeblendTask,
+            "meas_deblender": measDeblender.SourceDeblendTask,
+        }
+
+        # Ensure the deblend target aligns with the selected deblender.
+        if self.deblend.target != (expected := targetMap.get(self.deblender)):
+            raise ValueError(
+                f"Invalid target for '{self.deblender}': expected {expected}, got {self.deblend.target}"
+            )
+
+
+class MeasurementDriverBaseTask(pipeBase.Task):
+    """Base class for the mid-level driver running detection, deblending
+    (optional), and measurement algorithms in one go.
+
+    This driver simplifies the process of applying a small set of measurement
+    algorithms to images by abstracting away schema and table boilerplate. It
+    is particularly suited for simple use cases, such as processing images
+    without neighbor-noise-replacement or extensive configuration.
```

Comment:
This *neighbor-noise-replacement* part was added based on the ticket description, but I’m not sure why it’s such a big deal for the driver task. It can be turned on or off easily with `config.measurement.doReplaceWithNoise`.